### PR TITLE
Add ComposingNormalizer for NFC, NFKC, and UTS 46

### PR DESCRIPTION
**Pull request by hsivonen**
Also addresses #1994.

### CODE REVIEW

File: experimental/normalizer/src/lib.rs [hunk 1]
```diff
@@ -827,101 +1419,174 @@ impl DecomposingNormalizer {
...
-        self.normalize_iter(text.chars()).collect()
-    }
+    normalizer_methods!();
+}
 
-    /// Check whether a string slice is normalized.
-    pub fn is_normalized(&self, text: &str) -> bool {
-        self.normalize_iter(text.chars()).eq(text.chars())
-    }
+/// A normalizer for performing decomposing normalization.
```
Comment (echeran) [inline experimental/normalizer/src/lib.rs:1429]:
```suggestion
/// A normalizer for performing composing normalization.
```

File: experimental/normalizer/src/tests.rs [hunk 2]
```diff
@@ -58,4 +61,254 @@ fn test_uts46d_basic() {
...
+    assert_eq!(normalizer.normalize("a\u{0308}"), "ä");
+    assert_eq!(normalizer.normalize("A\u{0308}"), "ä");
+    assert_eq!(normalizer.normalize("e\u{0323}\u{0302}"), "ệ");
+    assert_eq!(normalizer.normalize("E\u{0323}\u{0302}"), "ệ");
+    assert_eq!(normalizer.normalize("𝅗𝅥"), "𝅗\u{1D165}"); // Composition exclusion
+
+    assert_eq!(normalizer.normalize("\u{2126}"), "ω"); // ohm sign
+    assert_eq!(normalizer.normalize("ﾍﾞ"), "ベ"); // half-width to full-width, the compose
+    assert_eq!(normalizer.normalize("ﬁ"), "fi"); // ligature expanded
+    assert_eq!(normalizer.normalize("\u{FDFA}"), "\u{635}\u{644}\u{649} \u{627}\u{644}\u{644}\u{647} \u{639}\u{644}\u{64A}\u{647} \u{648}\u{633}\u{644}\u{645}");
```
Comment (echeran) [inline experimental/normalizer/src/tests.rs:126]:
```suggestion
    assert_eq!(normalizer.normalize("\u{FDFA}"), "\u{0635}\u{0644}\u{0649} \u{0627}\u{0644}\u{0644}\u{0647} \u{0639}\u{0644}\u{064A}\u{0647} \u{0648}\u{0633}\u{0644}\u{0645}");
```

**Review by echeran — APPROVED** [review]
I left minor comments and suggestions, PTAL, otherwise LGTM.

**Review by echeran — APPROVED** [review]

**Comment by hsivonen** [comment]
> I left minor comments and suggestions, PTAL, otherwise LGTM.

Thanks. Landed.

**Pull request merged**